contracts/abstracted_account.algo.ts

```diff
@@ -174,4 +213,31 @@ export class AbstractedAccount extends Contract {
     this.namedPlugins(name).delete();
     this.plugins(app).delete();
   }
+
+  /**
+   * Add a gate for an app, it can also be used for renewals
+   *
+   * @param app The app to add the gate for
+   * @param address The duration of the gate
+   * @param duration The duration of the gate
+   */
+
+  addGate(app: Application, address: string, duration: uint64): void {
```

Comment:
If you're on-board with just having this information in the plugins BoxMap, I think it'd be better to not have these specific methods. If someone wants to change the configuration, they can just remove and re-add the plugin

Reply:
Do we lose anything if we don't even require them to remove it first?

Reply:
I think just having them add/remove reduces complexity by having one less method in the contract to edit. I suppose you could have add just modify data if it already exists, but that does seem a bit dangerous/unclear to me

Reply:
`upsertGate`?

Reply:
Yeah now that I think about it that should be fine